### PR TITLE
Wrong preproccessor condition fix

### PR DESCRIPTION
**Pull request by RonEld**
fix for issue #696
Change #if defined(MBEDTLS_THREADING_PTHREAD)
to #if defined(MBEDTLS_THREADING_C)

### CODE REVIEW

File: library/x509_crt.c [hunk 1]
```diff
@@ -1169,7 +1169,7 @@ int mbedtls_x509_crt_parse_path( mbedtls_x509_crt *chain, const char *path )
     if( dir == NULL )
         return( MBEDTLS_ERR_X509_FILE_IO_ERROR );
 
-#if defined(MBEDTLS_THREADING_PTHREAD)
+#if defined(MBEDTLS_THREADING_C)
```
Comment (andresag01) [inline library/x509_crt.c]: Could you please add a comment in the matching `#endif` as in https://github.com/ARMmbed/mbedtls/pull/739/files#diff-7eb92ee9f705f88338240a484631c19cR1216?

**Comment by andresag01** [comment]
Does this change need a ChangeLog entry? Does it need to be backported to mbed TLS 2.1 and 1.3?

**Comment by RonEld** [comment]
Thanks for the reminder!
This fix needs to be backported to mbed TLS 1.3 and 2.1

File: library/x509_crt.c (hunk 1 above)
Comment (RonEld) [inline library/x509_crt.c]: ok

**Comment by sbutcher-arm** [comment]
Reviewed, approved, tested and merged into `development`.

**Pull request closed without merging**